packages/compartment-mapper/src/types.js

```diff
@@ -387,6 +396,18 @@ export {};
  * @property {SourceMapHook} [sourceMapHook]
  */
 
+/**
+ * @typedef CustomParser
+ * @property {ParserImplementation} parser Parser implementation
+ * @property {Language} language Language the parser applies to. Can be new or existing
+ * @property {string[]} extensions Extensions the language applies to
+ */
+
+/**
+ * @typedef {object} ExtraImportOptions
+ * @property {CustomParser[]} [parsers] Custom parser objects
+ */
```

Comment:
What would this be named if we also supported custom parsers for `archive.js`, `bundle.js`, and `import-archive.js`? Perhaps we would be better served avoiding a name for it by defining the `ImportOptions`, `ImportArchiveOptions`, and `ArchiveOptions` and just including `parsers` on the ones that implement them so far.

Reply:
Yeah, tbh the piecemeal definition of those options on the function definitions has been bothering me.  I'll do something here.